pkg/gameservers/portallocator_test.go

```diff
@@ -39,6 +39,37 @@ var (
 	n3 = corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node3", UID: "node3"}}
 )
 
+func TestEmptyPortPolicy(t *testing.T) {
+	t.Parallel()
+	fixture := dynamicGameServerFixture()
+
+	t.Run("test portPolicy as an empty string", func(t *testing.T) {
+		m := agtesting.NewMocks()
+		pa := NewPortAllocator(10, 50, m.KubeInformerFactory, m.AgonesInformerFactory)
+		nodeWatch := watch.NewFake()
+		m.KubeClient.AddWatchReactor("nodes", k8stesting.DefaultWatchReactor(nodeWatch, nil))
+
+		stop, cancel := agtesting.StartInformers(m, pa.nodeSynced)
+		defer cancel()
+
+		// Make sure the add's don't corrupt the sync
+		// (no longer an issue, but leave this here for posterity)
+		nodeWatch.Add(&n1)
+		nodeWatch.Add(&n2)
+		assert.True(t, cache.WaitForCacheSync(stop, pa.nodeSynced))
+
+		err := pa.syncAll()
+		assert.Nil(t, err)
```

Comment:
Use require here, please. If an error happens here there is no point to continue.